src/core/lib/gprpp/time.h

```diff
@@ -61,6 +65,32 @@ class Duration;
 // Timestamp represents a discrete point in time.
 class Timestamp {
  public:
+  // Base interface for time providers.
+  class Source {
+   public:
+    // Return the current time.
+    virtual Timestamp Now() = 0;
+    virtual void InvalidateCache() {}
+
+   protected:
+    ~Source() = default;
```

Comment:
The dtor should be `virtual`.

Reply:
No, this is right. We make it protected so it can't be called from the base class (which is what the virtual-ness is for).

Reply:
Okay.  Please document why it's safe for this to be non-virtual in this case.

Reply:
done